src/components/auth/Register.jsx

```diff
@@ -1,9 +1,332 @@
+    if (password !== confirmPassword) {
+      setConfirmPasswordError('Passwords do not match');
+      isDataValid = false;
+    } else {
+      setConfirmPasswordError('');
+    }
+
+    if (isAdultNameRequired && adultName.trim() === '') {
+      setAdultNameError('Adult name is required for students under 16');
+      isDataValid = false;
+    } else {
+      setAdultNameError('');
+    }
+
+    if (isDataValid) {
+      console.log('SUBMIT FORM');
+      console.log(`Create ${userRole} account`);
+
+      try {
+        let result;
+        if (userRole == 'student') {
+          let studentData = {
+            "firstName": firstName.trim(),
+            "lastName": lastName.trim(),
+            "email": email.trim(),
+            "password": password,
+            "dateOfBirth": dateOfBirth,
+            "adultName": adultName.trim(),
+          };
+          result = await registerStudent(studentData);
+        } else {
+          let teacherData = {
+            "firstName": firstName,
+            "lastName": lastName,
+            "email": email,
+            "password": password,
+          };
+          result = await registerTeacher(teacherData);
+        }
+        console.log(result);
+        if (result.status === 201) {
+          setIsModalOpen(true);
+        }
+      } catch (error) {
+        console.log('ERROR');
+        console.log(error);
+        if (error.message.includes("taken")) {
+          setEmailError(error.message);
+        }
+      }
+    }
+
+
+  }
+
+  const switchUserRole = () => {
+    if (userRole === 'student') {
+      setUserRole('teacher');
+    } else {
+      setUserRole('student');
+    }
+  }
+
+  const handleFirstNameChange = (event) => {
+    const newFirstName = event.target.value;
+    setFirstName(newFirstName);
+  }
+
+  const handleLastNameChange = (event) => {
+    const newLastName = event.target.value;
+    setLastName(newLastName);
+  }
+
+  const handleDateOfBirthChange = (event) => {
+    const newBirthDate = event.target.value;
+    setDateOfBirth(newBirthDate);
+    const age = calculateAge(newBirthDate);
+    setIsAdultNameRequired(age < 16);
+    setAdultNameError('');
+  };
+
+  const handleAdultNameChange = (event) => {
+    setAdultName(event.target.value);
+  };
+
+  const handleEmailChange = (event) => {
+    const newEmail = event.target.value;
+    setEmail(newEmail);
+    if (newEmail.trim() !== '') {
+      setEmailError('');
+    }
+  }
+
+  const handleConfirmEmailChange = (event) => {
+    setConfirmEmail(event.target.value);
+  }
+
+  const handlePasswordChange = (event) => {
+    const newPassword = event.target.value;
+    setPassword(newPassword);
+    if (newPassword.trim() !== '') {
+      setPasswordError('');
+    }
+  }
+
+  const handleConfirmPasswordChange = (event) => {
+    setConfirmPassword(event.target.value);
+  }
+
+  const handleGotoLogin = () => {
+    setIsModalOpen(false);
+    navigate('/login');
+  }
+
+  const calculateAge = (birthdate) => {
```

Comment:
Nice job putting this together! We might have to open a ticket to create a module that holds all date management logic since there are a few functions that call the Date method scattered throughout the app. But you can leave this here for now.

Reply:
ticket to create dateUtils file was created